### PR TITLE
Added index.js file to rewiev

### PR DESCRIPTION
**Pull request by chernetskyi8704**
### OOP exercise

[Demo](https://chernetskyi8704.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/chernetskyi8704/a-tiny-JS-world/tree/populate-world)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by chernetskyi8704** [comment]
Has completed self-checks.

**Comment by chernetskyi8704** [comment]
Comment for activity.

File: submissions/chernetskyi8704/Tiny JS World/index.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  constructor(species, name, gender, saying) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.saying = saying;
+  }
+
+  displayData() {
+    return ["species", "name", "gender", "saying"]
+      .map((data) => this[data])
```
Comment (OleksiyRudenko) [inline submissions/chernetskyi8704/Tiny JS World/index.js:19]: `data` is a generic term. What exactly `map` works with?

File: submissions/chernetskyi8704/Tiny JS World/index.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+class Person extends Inhabitant {
+  constructor(species, name, gender, saying, hands, legs) {
+    super(species, name, gender, saying);
+    this.hands = hands;
+    this.legs = legs;
+  }
+
+  displayData() {
+    return `${super.displayData()}; ${this.hands}; ${this.legs};`;
```
Comment (OleksiyRudenko) [inline submissions/chernetskyi8704/Tiny JS World/index.js:32]: Please make this (and other similar fragments in subclasses) DRYer with `Array#join`.
Do not use strings representing propety names. KISS

File: submissions/chernetskyi8704/Tiny JS World/index.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+class Man extends Person {}
+class Woman extends Person {}
+class Dog extends Animal {}
+class Cat extends Animal {}
+
+const man = new Man("Man", "Dimitrij", "male", "Nice to meet you!", 2, 2);
+const woman = new Woman("Woman", "Kate", "male", "Have a good day!", 2, 2);
+const dog = new Dog("Dog", "Molly", "female", "WOOF-WOOF!", 4, 1);
+const cat = new Cat("Cat", "Whiskey", "female", "Meow!", 4, 1);
```
Comment (OleksiyRudenko) [inline submissions/chernetskyi8704/Tiny JS World/index.js:56]: From the checklist:
> No need to specify species at instantiation

There is at least two ways to achieve this. One you already use for a different property. Another one is hinted in the checklist.

File: submissions/chernetskyi8704/Tiny JS World/index.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+// ======== OUTPUT ========
+const allInhabitants = [man, woman, dog, cat];
+
+allInhabitants.map((inhabitan) => {
```
Comment (OleksiyRudenko) [inline submissions/chernetskyi8704/Tiny JS World/index.js:61]: Check English spelling.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@chernetskyi8704 quite a decent job!
A couple of improvements are still needed.

**Comment by OleksiyRudenko** [comment]
I just noticed there are [two submissions for TJSW OOP from you](https://github.com/kottans/frontend-2022-homeworks/pulls?q=is%3Apr+author%3Achernetskyi8704+label%3Atask-TJSW-OOP).

What tasks exactly implements each of #385 and #575?

They review flow is different for pre-OOP and post-OOP.

**Comment by chernetskyi8704** [comment]
In #385 I perhaps make more than it requires in 'Building a Tiny JS World (pre-OOP) - practice'. First, it was written like this - https://github.com/chernetskyi8704/a-tiny-JS-world/commit/dcab8f627ce4b788f6e04382ddad661b647fc9fc. But then I asked in the chat if it ok and replied that I needed to be made some changes. My bad that I incorrectly read the requirements.

**Comment by chernetskyi8704** [comment]
Should I replace this code with the code that in the commit that I send above or just make improvements wich you pointed on?

**Comment by OleksiyRudenko** [comment]
Hold on. I will suggest the approach so that we do not discard what's been done so far in terms of both code and review.

**Comment by chernetskyi8704** [comment]
Is there something requier to do from me?

File: submissions/chernetskyi8704/Tiny JS World/index.js (hunk 1 above)
Comment (chernetskyi8704) [reply submissions/chernetskyi8704/Tiny JS World/index.js]: I will take into account in the future, not to use a generic term in situations like that. Thanks a lot.

File: submissions/chernetskyi8704/Tiny JS World/index.js (hunk 2 above)
Comment (chernetskyi8704) [reply submissions/chernetskyi8704/Tiny JS World/index.js]: Thanks a lot. Now really code looks better.

File: submissions/chernetskyi8704/Tiny JS World/index.js (hunk 3 above)
Comment (chernetskyi8704) [reply submissions/chernetskyi8704/Tiny JS World/index.js]: Think now i made it correctly.

File: submissions/chernetskyi8704/Tiny JS World/index.js (hunk 4 above)
Comment (chernetskyi8704) [reply submissions/chernetskyi8704/Tiny JS World/index.js]: Sorry for that one, it was obvious.

**Comment by OleksiyRudenko** [comment]
@chernetskyi8704 let's keep this PR for OOP version and #575 for pre-OOP version.

**Comment by chernetskyi8704** [comment]
@OleksiyRudenko I hope I understood you correctly.

File: submissions/chernetskyi8704/Tiny JS World/index.js [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+class Man extends Human {
+  constructor(Name, Gender, Saying) {
+    super("Man🤵", Name, Gender, Saying);
+  }
+}
+
+class Woman extends Human {
+  constructor(Name, Gender, Saying) {
+    super("Woman👩", Name, Gender, Saying);
```
Comment (OleksiyRudenko) [inline submissions/chernetskyi8704/Tiny JS World/index.js:86]: Men and women belong to the same species. I just cannot let heresy pass through :)

File: submissions/chernetskyi8704/Tiny JS World/index.js [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+      this.saying()
+    );
+  }
+}
+
+class Cat extends Animal {
+  constructor(Name, Gender, Saying) {
+    super("Cat🐱", Name, Gender, Saying);
+    this.Paws = "4";
+    this.Tail = "1";
```
Comment (OleksiyRudenko) [inline submissions/chernetskyi8704/Tiny JS World/index.js:119]: Not critical here, as we do not really do math operations with paws and tails.
Just wondering, why strings?

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@chernetskyi8704 nearly there

File: submissions/chernetskyi8704/Tiny JS World/index.js (hunk 5 above)
Comment (chernetskyi8704) [reply submissions/chernetskyi8704/Tiny JS World/index.js:86]: Yeah, that was a stupid one.

File: submissions/chernetskyi8704/Tiny JS World/index.js (hunk 6 above)
Comment (chernetskyi8704) [reply submissions/chernetskyi8704/Tiny JS World/index.js:119]: Strings because of using string methods. 
Should i fix it?

Comment (OleksiyRudenko) [reply submissions/chernetskyi8704/Tiny JS World/index.js:119]: Not necessarily. For example, telephone numbers do not actually make sense as numbers in mathematial sense, as we do not math operations over them. From the other hand we may want to count total number of legs in a herd.

Probably a missing part in your puzzle was that numbers coerce to strings automagically in string context.
Check what `4 + ""` results in and why `[1, 31, 3, 5, 11].sort()` doesn't work as expected. This is an important part of understanding JS internals and peculiarities.

You may skip fixing this. Make sure to take some learnings.

Comment (chernetskyi8704) [reply submissions/chernetskyi8704/Tiny JS World/index.js:119]: Type conversion it's really a great thing in JS but I'm not thought about this in the border of this task. In the future will ask my self more often 'What about if...'. Thank you so much for the review.

**Review by OleksiyRudenko — APPROVED** [review]
@chernetskyi8704 great job!

**Pull request merged**